compiler/passes/scopeResolve.cpp

```diff
@@ -1722,10 +1722,14 @@ static bool lookupThisScopeAndUses(const char*           name,
               BaseAST* scopeToUse = use->getSearchScope();
 
               Symbol* sym = inSymbolTable(nameToUse, scopeToUse);
-              if (!sym) {
+              if (!sym && use->canReexport) {
                 if (ResolveScope* rs = ResolveScope::getScopeFor(scopeToUse)) {
                   sym = rs->lookupPublicUnqualAccessSyms(nameToUse, context,
                                                          renameLocs);
+                  // propagate this information to the UseStmt
+                  if (!rs->canReexport) {
+                    use->canReexport = false;
```

Comment:
Does this UseStmt check really help?  The lookupPublicUnqualAccessSyms already exits out early

Reply:
Yes. It wasn't in my initial plans either. And I think it makes the implementation a tiny bit uglier, but the getScopeFor call has non-negligible overhead. Adding that check makes a clear performance difference. This branch is executed way too many times that even if it is not doing something complicated (map.find, I think) it adds up.

Reply:
Okay, sounds good!